src/SimpleGraphs/generators/randgraphs.jl

```diff
@@ -237,40 +253,54 @@ julia> watts_strogatz(10, 4, 0.3)
 
 julia> watts_strogatz(Int8(10), 4, 0.8, is_directed=true, seed=123)
 {10, 20} directed simple Int8 graph
+
+### References
+- Collective dynamics of ‘small-world’ networks, Duncan J. Watts, Steven H. Strogatz. [https://doi.org/10.1038/30918](https://doi.org/10.1038/30918)
+- Small Worlds, Duncan J. watts. [https://en.wikipedia.org/wiki/Special:BookSources?isbn=978-0691005416](https://en.wikipedia.org/wiki/Special:BookSources?isbn=978-0691005416)
 ```
 """
 function watts_strogatz(n::Integer, k::Integer, β::Real; is_directed=false, seed::Int=-1)
+
     @assert k < n
-    if is_directed
-        g = SimpleDiGraph(n)
-    else
-        g = SimpleGraph(n)
+
+    if k == n - 1 && iseven(k)
+        return is_directed ? complete_digraph(n) : complete_graph(n)
     end
+
+    g = is_directed ? SimpleDiGraph(n) : SimpleGraph(n)
+
     rng = getRNG(seed)
-    for s in 1:n
-        for i in 1:(floor(Integer, k / 2))
-            target = ((s + i - 1) % n) + 1
-            if rand(rng) > β && !has_edge(g, s, target) # TODO: optimize this based on return of add_edge!
-                add_edge!(g, s, target)
-            else
-                while true
-                    d = target
-                    while d == target
-                        d = rand(rng, 1:(n - 1))
-                        if s < d
-                            d += 1
-                        end
-                    end
-                    if s != d
-                        add_edge!(g, s, d) && break
-                    end
-                end
-            end
+
+    target(s, i) = ((s + i - 1) % n) + 1
```

Comment:
we generally don't define nested functions, but I think this might be ok. Could you add a comment here describing why this works?

Reply:
Sure! (Just reusing `target = ((s + i - 1) % n) + 1` from the original, really. Made it a function, since I'm using it twice.)

Reply:
Might want to put an `@inline` in there then just to help the compiler out.